roles/openshift_health_checker/openshift_checks/etcd_volume_check.py

```diff
@@ -0,0 +1,68 @@
+# pylint: disable=missing-docstring
+"""
+Ansible module for determining information about etcd volume size.
+"""
+
+from openshift_checks import OpenShiftCheck, get_var
+
+
+class EtcdImageData(OpenShiftCheck):
+    """Check that etcd volume capacity is not over 50%"""
+
+    name = "etcd_volume_check"
+    tags = ["etcd_volume_check"]
+
+    def run(self, _, task_vars):
+        ansible_mounts = get_var(task_vars, "ansible_mounts")
+        etcd_mountpath = self.get_etcd_mountpath(ansible_mounts)
+        etcd_diskspace_avail = self.get_etcd_diskspace(etcd_mountpath, ansible_mounts)
+        etcd_totaldiskspace = self.get_etcd_totaldiskspace(etcd_mountpath, ansible_mounts)
```

Comment:
To use the values from the `setup` module you will need to ensure they are passed into the module invocation from the action plugin. Another option, which is probably a bit cleaner would be to leverage `module_utils` to re-gather the needed facts at runtime: https://github.com/ansible/ansible/blob/v2.2.1.0-1/lib/ansible/module_utils/facts.py